include/seqan3/io/sequence/sequence_file_in.hpp

```diff
@@ -1,175 +1,911 @@
+                  "You selected a field that is not valid for sequence files, please refer to the documentation "
+                  "of sequence_file_in::field_ids for the accepted values.");
+
+    static_assert([] () constexpr
+                  {
+                      return !(selected_field_ids::contains(field::SEQ_QUAL) &&
+                               (selected_field_ids::contains(field::SEQ) ||
+                               (selected_field_ids::contains(field::QUAL))));
+                  }(),
+                  "You may not select field::SEQ_QUAL and either of field::SEQ and field::QUAL at the same time.");
+
+    /*!\name Field types and record type
+     * \brief These types may be manipulated via the \ref traits_type to achieve different storage behaviour.
+     * \{
+     */
+    //!\brief The type of field::SEQ (std::vector <seqan3::dna5> by default).
+    using sequence_type         = typename traits_type::template sequence_container<
+                                    typename traits_type::sequence_alphabet>;
+    //!\brief The type of field::ID (std::string by defaul).
+    using id_type               = typename traits_type::template id_container<
+                                    typename traits_type::id_alphabet>;
+    //!\brief The type of field::QUAL (std::vector <seqan3::phred42> by default).
+    using quality_type          = typename traits_type::template quality_container<
+                                    typename traits_type::quality_alphabet>;
+    //!\brief The type of field::SEQ_QUAL (std::vector <seqan3::dna5q> by default).
+    using sequence_quality_type = typename traits_type::
+                                    template sequence_container<quality_composition<typename traits_type::sequence_alphabet,
+                                                                                    typename traits_type::quality_alphabet>>;
 
-    // constructor with arg
-    sequence_file_in(std::string const & _file_name);
+    //!\brief The previously defined types aggregated in a seqan3::type_list.
+    using field_types       = type_list<sequence_type, id_type, quality_type, sequence_quality_type>;
 
-    // copy construction and assignment are deleted
-    // implicitly because we don't want multiple access to file
+    //!\brief The type of the record, a specialisation of seqan3::record; acts as a tuple of the selected field types.
+    using record_type       = record<detail::select_types_with_ids_t<field_types, field_ids, selected_field_ids>,
+                                     selected_field_ids>;
+    //!\}
+
+    /*!\name Field column types
+     * \brief These types may be manipulated via the \ref traits_type to achieve different storage behaviour.
+     * \{
+     */
+    //!\brief Column type of field::SEQ; usually seqan3::concatenated_sequences<sequence_type>.
+    using sequence_column_type          = typename traits_type::template sequence_container_container<sequence_type>;
```

Comment:
I am not too happy with it either, but your proposed replacements are even worse IMHO. `sequence_range` is just strange, because a sequence is a range...

I plan to add more documentation of the general "row VS column" thing to the overall i/o documentation, but it's not done, yet. After that "column" will be more clear, I think.